src/ImageSearchModal.jsx

```diff
@@ -15,238 +15,203 @@ import cockpit from 'cockpit';
+export const ImageSearchModal = ({ downloadImage, registries, user, userServiceAvailable, systemServiceAvailable }) => {
+    const [searchInProgress, setSearchInProgress] = useState(false);
+    const [searchFinished, setSearchFinished] = useState(false);
+    const [imageIdentifier, setImageIdentifier] = useState('');
+    const [imageList, setImageList] = useState([]);
+    const [imageTag, setImageTag] = useState("");
+    const [isSystem, setIsSystem] = useState(systemServiceAvailable);
+    const [selectedRegistry, setSelectedRegistry] = useState("");
+    const [selected, setSelected] = useState("");
+    const [dialogError, setDialogError] = useState("");
+    const [dialogErrorDetail, setDialogErrorDetail] = useState("");
+    const [typingTimeout, setTypingTimeout] = useState(null);
+
+    let activeConnection = null;
+    const Dialogs = useDialogs();
+    // Registries to use for searching
+    const searchRegistries = registries.search && registries.length !== 0 ? registries.search : fallbackRegistries;
+
+    // Don't use on selectedRegistry state variable for finding out the
+    // registry to search in as with useState we can only call something after a
+    // state update with useEffect but as onSearchTriggered also changes state we
+    // can't use that so instead we pass the selected registry.
+    const onSearchTriggered = (searchRegistry = "", forceSearch = false) => {
         // When search re-triggers close any existing active connection
-        if (this.activeConnection)
-            this.activeConnection.close();
-        this.setState({ searchFinished: false });
+        activeConnection = rest.connect(client.getAddress(isSystem), isSystem);
+        if (activeConnection)
+            activeConnection.close();
+        setSearchFinished(false);
 
         // Do not call the SearchImage API if the input string  is not at least 2 chars,
         // unless Enter is pressed, which should force start the search.
         // The comparison was done considering the fact that we miss always one letter due to delayed setState
-        if (this.state.imageIdentifier.length < 2 && !forceSearch)
+        if (imageIdentifier.length < 2 && !forceSearch)
             return;
 
-        this.setState({ searchInProgress: true });
+        setSearchInProgress(true);
 
-        this.activeConnection = rest.connect(client.getAddress(this.state.isSystem), this.state.isSystem);
-        let registries = Object.keys(this.props.registries).length !== 0 ? [this.state.registry] : fallbackRegistries;
+        let queryRegistries = searchRegistries;
+        if (searchRegistry !== "") {
+            queryRegistries = [searchRegistry];
+        }
         // if a user searches for `docker.io/cockpit` let podman search in the user specified registry.
-        if (this.state.imageIdentifier.includes('/')) {
-            registries = [""];
+        if (imageIdentifier.includes('/')) {
+            queryRegistries = [""];
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230705-092041-36f39867-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)